spec.emu

```diff
@@ -45,14 +45,15 @@ contributors: Chengzhong Wu, Hemanth HM, Ecma International
 
     <emu-clause id="sec-createnonenumerabledataproperty">
       <h1><ins>CreateNonEnumerableDataProperty ( _O_, _P_, _V_ )</ins></h1>
-      <p><ins>The abstract operation CreateNonEnumerableDataProperty takes arguments _O_ (an Object), _P_ (a property key), and _V_ (an ECMAScript language value). It performs the following steps when called:</ins></p>
+      <p><ins>The abstract operation CreateNonEnumerableDataProperty takes arguments _O_ (an Object), _P_ (a property key), and _V_ (an ECMAScript language value). It is used to create a new non-enumerable own property of an object. It performs the following steps when called:</ins></p>
       <emu-alg>
         1. <ins>Assert: Type(_O_) is Object.</ins>
-        1. <ins>Assert: IsPropertyKey(_P_) is true.</ins>
+        1. <ins>Assert: IsPropertyKey(_P_) is *true*.</ins>
         1. <ins>Let _newDesc_ be the PropertyDescriptor { [[Value]]: _V_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</ins>
-        1. <ins>Return ? DefinePropertyOrThrow(_O_, _P_, _newDesc_).</ins>
+        1. <ins>Return ? ? _O_.[[DefineOwnProperty]](_P_, _newDesc_).</ins>
```

Comment:
Why is this an improvement? The behavior of returning false on failure has caused a number of spec bugs, which is why most of them have been changed to call DefinePropertyOrThrow with `?`.

Reply:
Based on the linked comment, the AO should have an “OrThrow” in the name, rather than not throwing at all.